src/script/components/asset/AudioAsset.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useEffect, useState} from 'react';
+import {TeamState} from '../../team/TeamState';
+import {container} from 'tsyringe';
+import cx from 'classnames';
+
+import {getLogger} from 'Util/Logger';
+import {formatSeconds} from 'Util/TimeUtil';
+
+import {AssetTransferState} from '../../assets/AssetTransferState';
+import type {ContentMessage} from '../../entity/message/ContentMessage';
+import type {FileAsset} from '../../entity/message/FileAsset';
+
+import RestrictedAudio from './RestrictedAudio';
+import SeekBar from './controls/SeekBar';
+import AudioSeekBar from './controls/AudioSeekBar';
+import AssetHeader from './AssetHeader';
+import MediaButton from './controls/MediaButton';
+
+import Icon from 'Components/Icon';
+import useEffectRef from 'Util/useEffectRef';
+import {useAssetTransfer} from './AbstractAssetTransferStateTracker';
+import {AssetRepository} from '../../assets/AssetRepository';
+import {registerReactComponent} from 'Util/ComponentUtil';
+
+const logger = getLogger('AudioAssetComponent');
+
+export interface AudioAssetProps {
+  assetRepository?: AssetRepository;
+  className?: string;
+  header: boolean;
+  // Does the asset have a visible header?
+  message: ContentMessage;
+  teamState?: TeamState;
+}
+
+const AudioAsset: React.FC<AudioAssetProps> = ({
+  header,
+  message,
+  className,
+  teamState = container.resolve(TeamState),
+  assetRepository = container.resolve(AssetRepository),
+}) => {
+  const asset = message.getFirstAsset() as FileAsset;
+  const [audioElement, setAudioElement] = useEffectRef<HTMLAudioElement>();
+  const isFileSharingReceivingEnabled = teamState.isFileSharingReceivingEnabled();
+  const {transferState, uploadProgress, cancelUpload} = useAssetTransfer(message);
```

Comment:
`useAssetTransfer` returns a function `loadAsset` which can be used in line 78. This way we don't need to inject the assetRepository anymore.

Reply:
https://github.com/wireapp/wire-webapp/pull/11577/commits/fbf7479538dc1209f6457934b3b5de970eef0b07